### PR TITLE
Enable pipelining during test

### PR DESCRIPTION
**Pull request by xwjiang-ms**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
When we enable fips in image, test time will increase a lot because of new method of setting up ssh connection. 
#### How did you do it?
Enable pipelining try to re-use existing SSH connections. Overhead of setting up new SSH connection could be saved, which will help us save about 4 hours
Use "docker exec -t" in tests, fix allocate tty issue.
#### How did you verify/test it?
Run a round of test

### CODE REVIEW

**Comment by wangxin** [comment]
Statement `multiple ssh connections may be created in a test` may not be true. If this ansible configuration is enabled, ansible will try to re-use existing SSH connections. Overhead of setting up new SSH connection could be saved.

**Comment by xwjiang-ms** [comment]
OK, I will change the description

**Review by wangxin — APPROVED** [review]

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/arp/test_arp_dualtor.py:12:1: F401 'tests.common.dualtor.mux_simulator_control.toggle_all_simulator_ports_to_upper_tor' imported but unused<br/>tests/arp/test_arp_dualtor.py:13:1: F401 'tests.common.dualtor.dual_tor_utils.upper_tor_host' imported but unused<br/>tests/arp/test_arp_dualtor.py:13:1: F401 'tests.common.dualtor.dual_tor_utils.lower_tor_host' imported but unused<br/>tests/arp/test_arp_dualtor.py:13:1: F401 'tests.common.dualtor.dual_tor_utils.config_dualtor_arp_responder' imported but unused<br/>tests/arp/test_arp_dualtor.py:13:121: E501 line too long (156 > 120 characters)<br/>tests/arp/test_arp_dualtor.py:14:1: F401 'tests.common.dualtor.dual_tor_common.mux_config' imported but unused<br/>tests/arp/test_arp_dualtor.py:15:1: F401 'tests.common.fixtures.ptfhost_utils.run_garp_service' imported but unused<br/>tests/arp/test_arp_dualtor.py:15:1: F401 'tests.common.fixtures.ptfhost_utils.change_mac_addresses' imported but unused<br/>tests/arp/test_arp_dualtor.py:15:1: F401 'tests.common.fixtures.ptfhost_utils.run_icmp_responder' imported but unused<br/>tests/arp/test_arp_dualtor.py:15:1: F401 'tests.common.fixtures.ptfhost_utils.pause_garp_service' imported but unused<br/>tests/arp/test_arp_dualtor.py:15:121: E501 line too long (126 > 120 characters)<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Review by wangxin — APPROVED** [review]

File: tests/dhcp_relay/test_dhcp_relay.py [hunk 1]
```diff
@@ -240,7 +240,7 @@ def testing_config(request, duthosts, rand_one_dut_hostname, tbinfo):
 
 
 def check_interface_status(duthost):
-    if ":67" in duthost.shell("docker exec -it dhcp_relay ss -nlp | grep dhcrelay",
+    if ":67" in duthost.shell("docker exec -i dhcp_relay ss -nlp | grep dhcrelay",
```
[the remark this answers is not in the document]
Comment (xwjiang-ms) [reply tests/dhcp_relay/test_dhcp_relay.py]: Oh, right

**Pull request merged**